.github/workflows/tests_gpu_python.yml

```diff
@@ -150,13 +150,21 @@ jobs:
         run: |
           python -m pip install cutensornet-cu${{ matrix.cuda_version }}
 
-      - name: Checkout PennyLane for release build
-        if: inputs.pennylane-version == 'release'
+      - name: Checkout PennyLane for release or latest build
+        if: inputs.pennylane-version == 'release' || inputs.pennylane-version == 'latest'
         uses: actions/checkout@v4
         with:
           path: pennylane
           repository: PennyLaneAI/pennylane
 
+      - name: Switch to latest build of PennyLane
+        if: inputs.pennylane-version == 'latest'
+        run: |
+          cd pennylane
+          git fetch origin master
+          git checkout master
```

Comment:
Actually now that I think about it this might be unnecessary. The `actions/checkout@v4` action checks out the master branch by default, so we don't need to `git fetch` and `git checkout` as we should already be on the master branch. We need to `fetch --all` for the release build because we need to first find the RC branch.

Reply:
I suggest to keep `git checkout master` to make clear what is going on. I will remove `git fetch origin master`

Reply:
Good idea :)